include/onnxruntime/core/common/common.h

```diff
@@ -283,4 +283,10 @@ inline std::wstring ToWideString(const std::wstring& s) { return s; }
 inline std::string ToWideString(const std::string& s) { return s; }
 #endif
 
+#if ((__cplusplus >= 201703L) || (defined(_MSVC_LANG) && (_MSVC_LANG >= 201703L)))
+#define IF_CONSTEXPR if constexpr
+#else
+#define IF_CONSTEXPR if
```

Comment:
>IF_CONSTEXPR [](http://example.com/codeflow?start=8&length=12)

why do we need a macro for this?

Reply:
Because CUDA 10.2 doesn't support C++17 and we still need to generate CUDA 10.2 packages, please avoid to use C++17 in CUDA files(*.cu). Also, the *.h files that they include(like core/framework/float16.h). In these *.h files, we need to use such a macro.

Reply:
ok, can we name it `ORT_IF_CONSTEXPR`?

Reply:
Done.